app.py

```diff
@@ -15,57 +12,40 @@
 
 class Thing(Resource):
     def get(self, name):
-        # TODO: do not interact with the DAO from the APP, the APP will communicate with the MODEL to handle all business logic
-        thing = dao.find(name)
-        
+        thing = Model.get_by_name(name)
         if not thing:
             abort(404, message='Thing {} not found'.format(name))
-            
-        return thing.serialize()
+        return vars(thing)
 
     def delete(self, name):
-        # TODO: do not interact with the DAO from the APP, the APP will communicate with the MODEL to handle all business logic
-        dao.delete(name)
-        
-        return name, 204
+        thing = Model.delete(name)
+        if not thing:
+            abort(404, message='Thing {} not found'.format(name))
+        return vars(thing), 204
 
     def put(self, name):
         args = parser.parse_args()
-        thing = dao.delete(name)
+        thing = Model.update(name, args['function'])
         if not thing:
             abort(404, message='Thing {} not found'.format(name))
-        
-        # TODO: do not interact with the DAO from the APP, the APP will communicate with the MODEL to handle all business logic
-        dao.add(Model.from_dict({'name':name, 'function':args['function']}))
-
-        return name, 201
-
+        return vars(thing), 201
 
 class ThingList(Resource):
     def get(self):
-        thing_list = list(dao.find_all())
+        thing_list = list(Model.find_all())
```

Comment:
these should be within a try/except

Reply:
the model should also return a list of thing objects .. that will just be `jsonified` and returned to the user